pkg/plugins/builtin/script/script.go

```diff
@@ -0,0 +1,108 @@
+package script
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	gexec "os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/ovh/utask/pkg/plugins/taskplugin"
+)
+
+// the script plugin execute scripts
+var (
+	Plugin = taskplugin.New("script", "0.1", exec,
+		taskplugin.WithConfig(validConfig, Config{}),
+	)
+)
+
+// Config is the configuration needed to execute a script
+type Config struct {
+	File    string   `json:"file,required"`
+	Argv    []string `json:"argv,omitempty"`
+	Timeout string   `json:"timeout_seconds,omitempty"`
+}
+
+func validConfig(config interface{}) error {
+	cfg := config.(*Config)
+
+	if cfg.File == "" {
+		return errors.New("file is missing")
+	}
+
+	if cfg.Timeout != "" {
+		if _, err := strconv.ParseUint(cfg.Timeout, 10, 64); err != nil {
+			return fmt.Errorf("timeout is wrong %s", err.Error())
+		}
+	}
+
+	return nil
+}
+
+func exec(stepName string, config interface{}, ctx interface{}) (interface{}, interface{}, error) {
+	cfg := config.(*Config)
+
+	var timeout time.Duration
+
+	if cfg.Timeout != "" {
+		t, _ := strconv.ParseInt(cfg.Timeout, 10, 64)
+		timeout = time.Duration(t)
+	} else {
+		// default is 5*60 = 300 seconds or 5 minutes
+		timeout = time.Duration(300)
```

Comment:
default should probably be lower than 5 minutes

Reply:
but this is actually 300ms

Reply:
I switch to 2minutes. And it's not 300ms, you missed the conversion. I moved the conversion for more readability.

Reply:
I'd even go down to 30s